### ISSUE TITLE
get rid of restUriNoAPI in taskworker config

### ISSUE THREAD

**Issue by belforte** [dmwm/CRABServer#6065]
it is currently used (at least) in DBDataDiscovery
```
                       tapeRecallStatusSet = server.post(self.config.TaskWorker.restURInoAPI+'task', data = urllib.urlencode(configreq))
```
and surely a direct call to config. from there should not be done.

**Comment by belforte** [comment]
currently submissions fail with
```
020-05-29 23:08:28,535:ERROR:Handler:Problem handling 200529_210735:cpena_crab_Data_MuonEG_B_2_v1 because of (6, 
'Could not resolve host: crabserver') failure, traceback follows
Traceback (most recent call last):
  File "/data/srv/TaskManager/3.3.2005.rc5/slc7_amd64_gcc630/cms/crabtaskworker/3.3.2005.rc5/lib/python2.7/site-pa
ckages/TaskWorker/Actions/Handler.py", line 77, in executeAction
    output = work.execute(nextinput, task=self._task, tempDir=self.tempDir)
  File "/data/srv/TaskManager/3.3.2005.rc5/slc7_amd64_gcc630/cms/crabtaskworker/3.3.2005.rc5/lib/python2.7/site-pa
ckages/TaskWorker/Actions/DBSDataDiscovery.py", line 100, in execute
    result = self.executeInternal(*args, **kwargs)
  File "/data/srv/TaskManager/3.3.2005.rc5/slc7_amd64_gcc630/cms/crabtaskworker/3.3.2005.rc5/lib/python2.7/site-pa
ckages/TaskWorker/Actions/DBSDataDiscovery.py", line 275, in executeInternal
    tapeRecallStatusSet = server.post(self.config.TaskWorker.restURInoAPI+'task', data = urllib.urlencode(configre
q))
  File "/data/srv/TaskManager/3.3.2005.rc5/slc7_amd64_gcc630/cms/crabtaskworker/3.3.2005.rc5/lib/python2.7/site-pa
ckages/RESTInteractions.py", line 85, in post
    return self.makeRequest(uri = uri, data = data, verb = 'POST')
  File "/data/srv/TaskManager/3.3.2005.rc5/slc7_amd64_gcc630/cms/crabtaskworker/3.3.2005.rc5/lib/python2.7/site-pa
ckages/RESTInteractions.py", line 129, in makeRequest
    verbose=self['verbose'])
  File "/data/srv/TaskManager/3.3.2005.rc5/slc7_amd64_gcc630/cms/crabtaskworker/3.3.2005.rc5/lib/python2.7/site-pa
ckages/WMCore/Services/pycurl_manager.py", line 257, in request
    curl.perform()
error: (6, 'Could not resolve host: crabserver')


```

**Comment by belforte** [comment]
this appears due to my recent changes in the code.

**Comment by belforte** [comment]
indeed inside DBSDataDiscovery these are already available
```
self.server
self.restURInoAPI
```
as per
https://github.com/dmwm/CRABServer/blob/418ec57719220e0ae5ed40bcce143d5fda0d1727/src/python/TaskWorker/Actions/TaskAction.py#L11-L40

**Comment by belforte** [comment]
use a new issue #6066 for tracking and fixing the bug as int he code as a hot fix.
Some code cleanup may still be needed to get rid of abuse of restUIRInoApi

**Comment by belforte** [comment]
with the fix for #6066 all references are to `self.restURINoAPI` and there is no more any reference to it in the  configuration.
